src/main/java/com/impactupgrade/nucleus/client/SfdcClient.java

```diff
@@ -646,6 +646,92 @@ protected QueryResult querySmsContacts(String updatedSinceClause, String filter,
     return query(query);
   }
 
+  //TODO: getDonators? getDonatorContacts?
+  public List<QueryResult> getDonatingContacts(Calendar updatedSince)
+      throws ConnectionException, InterruptedException {
+    List<QueryResult> queryResults = new ArrayList<>();
+
+    String updatedSinceClause = "";
+    if (updatedSince != null) {
+      updatedSinceClause = "SystemModStamp >= " + new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").format(updatedSince.getTime());
+    }
+    queryResults.add(queryDonatingContacts(updatedSinceClause));
+    queryResults.add(queryDonatingAccountsContacts(updatedSinceClause));
+
+    if (updatedSince != null) {
+      updatedSinceClause = "Id IN (SELECT ContactId FROM CampaignMember WHERE SystemModStamp >= " + new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").format(updatedSince.getTime()) + ")";
+      queryResults.add(queryDonatingContacts(updatedSinceClause));
+    }
```

Comment:
This block can be removed -- the CampaignMember sync is pretty specific to the Mailchimp sync due to how SFDC Campaigns -> MC Tags work.